lib/core_ext/uri.rb

```diff
@@ -3,19 +3,22 @@
 
 module URI
   class Generic
+    # Avoid a deprecation warning with Ruby 3.4 where the default parser was changed to RFC3986
+    PARSER = T.let(const_defined?(:RFC2396_PARSER) ? RFC2396_PARSER : DEFAULT_PARSER, RFC2396_Parser)
```

Comment:
If the default changed, can we use the new default? Or is there some sort of incompatibility?

Reply:
You can't, there is no equivalent in the RFC3986 parser for this. https://github.com/ruby/uri/blob/be858ed90548425eb278cee2116947c7b749cb4e/lib/uri/rfc3986_parser.rb#L155-L160 I have no idea why it is this way.

I'll get this PR ready tomorrow.